react/features/remote-video-menu/actions.js

```diff
@@ -12,3 +27,56 @@ import { RemoteVideoMenu } from './components';
 export function hideRemoteVideoMenu() {
     return hideDialog(RemoteVideoMenu);
 }
+
+/**
+ * Mutes the local participant.
+ *
+ * @returns {Function}
+ */
+export function muteLocal() {
+    return (dispatch: Dispatch<any>) => {
+        const enable = true;
+
+        sendAnalytics(createToolbarEvent(AUDIO_MUTE, { enable }));
+        dispatch(setAudioMuted(enable, true));
+
+        // FIXME: The old conference logic as well as the shared video feature
+        // still rely on this event being emitted.
+        APP.UI.emitEvent(UIEvents.AUDIO_MUTED, enable, true);
+    };
+}
+
+/**
+ * Mutes the remote participant with the given ID.
+ *
+ * @param {string} participantId - ID of the participant to mute.
+ * @returns {Function}
+ */
+export function muteRemote(participantId: string) {
+    return (dispatch: Dispatch<any>) => {
+        sendAnalytics(createRemoteMuteConfirmedEvent(participantId));
+        dispatch(muteRemoteParticipant(participantId));
+    };
+}
+
+/**
+ * Mutes all participants.
+ *
+ * @param {Array<string>} exclude - Array of participant IDs to not mute.
+ * @returns {Function}
+ */
+export function muteAllParticipants(exclude: Array<string>) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const state = getState();
+        const localId = getLocalParticipant(state).id;
+        const participantIds = state['features/base/participants']
+            .map(p => p.id);
+
+        /* eslint-disable no-confusing-arrow */
+        participantIds
+            .filter(id => !exclude.includes(id))
+            .map(id => id === localId ? muteLocal() : muteRemote(id))
+            .map(dispatch);
```

Comment:
So in order to dispatch the everything we practically go over the participant array 4 times. Wouldn't this be better:
```
state['features/base/participants'].forEach(({ id }) => {
    if (!exclude.includes(id)) {
        dispatch(id === localId ? muteLocal() : muteRemote(id));
    }
});
```

IMHO looks simpler and goes over the participants array only ones. WDYT?

Reply:
I can change it to that if you like that more - but as a general idea, (1 iteration × 4 operations) === (4 iterations x 1 operation). There are more things going on here, but generally speaking.

Reply:
Actually good point! Now I agree it is probably not a big deal. It is your choice than! Whatever you prefer, just let me know!

Reply:
I'll leave it as is if its ok with you!